services/121-service/src/payments/imagecode/image-code.service.ts

```diff
@@ -107,6 +107,25 @@ export class ImageCodeService {
     return `${day} ${month} ${date.getFullYear()} ${time}`;
   }
 
+  private extendDecimalsAmount(
+    amount: string,
+    showZerosIfInteger: boolean,
+  ): string {
+    if (!amount.includes('.')) {
+      if (showZerosIfInteger) {
+        return `${amount}.00`;
+      } else {
+        return amount;
+      }
+    }
+    const amountParts = amount.split('.');
+    if (amountParts[1].length === 1) {
+      return `${amount}0`;
+    } else {
+      return `${amountParts[0]}.${amountParts[1].substring(0, 2)}`;
+    }
```

Comment:
There is also a `padEnd`-method: https://devdocs.io/javascript/global_objects/string/padend

So, `amountParts[1].padEnd(2, '0')`.


But I have the feeling there must be a simpler way to get a "decimal currency amount string" from a (floating) number...

Reply:
@arsforza thanks. good suggestion. I implemented it with some small edits.